EasyModbusSource/README.md

```diff
@@ -0,0 +1,286 @@
+
+
+
+## Select Statements
+
+In order to interact with the EasyModbus Source, one option is to use VAIL to select from the source. To do this, you will need to specify the SQL liked Query you wish to execute against your EasyModbus source. The data will be returned to Vantiq as a json buffer , which contains all fields of the different storages.
+
+used here would be **SELECT STATEMENTS**. The data will be returned to VANTIQ as a set of messages, where each message 
+contains some number of rows.
+
+There are 4 types of information you can select from the EasyModbusServer:  registers and holdingRegisters which return a list of integers , and discrete and coils which return a list of booleans. 
+
+you can use 1 of the 4 available select statments :
+```
+     select * from source EasyModbus1 as r with query : "select * from registers"
+ 	select * from source EasyModbus1 as r with query : "select * from holdingregisters"
+ 	select * from source EasyModbus1 as r with query : "select * from discrete"
+ 	select * from source EasyModbus1 as r with query : "select * from coils"
+```
+those select statments will return the entire list of registers, holdingRegisters, discrete or coils 
+```
+    select * from source EasyModbus1 as r with query : "select item3 from holdingregisters"
+```
+The above select statement will return the 4th holdingregister value indicating that fact using the index property 
+
+the following select statment will return a specific value from the requested list : 
+
+```
+[
+   {
+      "registers": [
+         {
+            "index": 3,
+            "value": 1
+         }
+      ]
+   }
+]
+```
+the following select statement will return a specific value from the requested list, actualy the first value in the colis list, noticed that the returned value is boolean and not integer as the previos example
+
+```
+select * from source EasyModbus1 as r with query : "select item0 from coils" 
+```
+
+```
+[
+   {
+      "values": [
+         {
+            "index": 0,
+            "value": true
+         }
+      ]
+   }
+]
+```
+
+## Publish Statements
+
+Using Publish statmet one can updates values in the PLC, EasyModbus server accept update request only for
+the holdingregisters and the coils list . 
+
+publish {"type":"holdingregisters"} to SOURCE EasyModbus1 using {body=r}
+
+where r is a json object similar to the result of the select statment , that above publish statement will update the entire list, one can use the property `address` in order to be able to update a specific value in the list, for example:
+```
+publish {"type":"holdingregisters",address:3} to SOURCE EasyModbus1 using {body=r}
+```
+will update only the 4th element using the value in the r json object, expecting the correct element with the right index value to exists - in this case 3. 
+
+the following statement will select the 4th value and update it with a different value, this can be shown using the server side simulator 
```

Comment:
```suggestion
The following statement will select the 4th value and update it with a different value, this can be shown using the server side simulator 
```

Reply:
done